Automattic/wp-calypso
# Framework: Transpile untranspiled node_modules, Take Two

By looking for an `esnext` or `module` field in the module's `package.json`. Allows us to retain ES6+ features such as tree shaking.
For background, see http://2ality.com/2017/06/pkg-esnext.html

Reverts Automattic/wp-calypso#19697, which reverted Take One: #16870. Revert was necessitated since #16870 failed to build. 

The build error https://github.com/Automattic/wp-calypso/pull/19697#issuecomment-343958188 due to which #16870 was failing was likely due to `find-root` being in `devDependencies` instead of `dependencies`, even tho it is required by `webpack.config.js`, and we're not installing `devDependencies` in Docker. (Props @sirreal for [pointing that out](https://github.com/Automattic/wp-calypso/pull/19698#discussion_r151362946) to me!)

To test:
* `npm run build-docker`
* `npm run docker`
* Have `wpcalypso.wordpress.com` point to `127.0.0.1` in your `/etc/hosts`
* Verify that `wpcalypso.wordpress.com` works (especially the notifications panel!)
